identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -33,34 +36,41 @@ public UserService(
     this.passwordEncoder = passwordEncoder;
   }
 
+  @Transactional
   public CamundaUser createUser(final CamundaUserWithPassword userWithCredential) {
     try {
       final UserDetails userDetails =
-          org.springframework.security.core.userdetails.User.withUsername(
-                  userWithCredential.user().username())
+          User.withUsername(userWithCredential.user().username())
               .password(userWithCredential.password())
               .passwordEncoder(passwordEncoder::encode)
               .disabled(!userWithCredential.user().enabled())
               .roles("DEFAULT_USER")
               .build();
       userDetailsManager.createUser(userDetails);
-      return new CamundaUser(userDetails.getUsername(), userDetails.isEnabled());
+      userRepository.createProfile(userWithCredential.user());
```

Comment:
❓ Instead of making two requests here (one to create the profile, and one to load the user) can we have a single method that creates (and returns) the user? One way I'm thinking is using the `returning` keyword in sql